frontend/frontend-react/src/components/RacksView.js

```diff
@@ -82,60 +87,169 @@ export class RacksView extends Component {
+      }
+    />
+
+    let empty;
+
+    if(!this.props.rack){
+      console.log('empty');
+      empty = 
+      <Grid item justify="flex-start" alignContent='center' xs={2}>
+        No racks.
+      </Grid>;
     }
 
+
+
     return (
-      <div>
-        <div id='hideOnPrint'>
-          <ButtonToolbar>
-            {adminTools}
-            <ButtonGroup>
-              <Button size="sm" onClick={this.handleCondensation}>Condensed Rack View</Button>{' '}
-              <Button size="sm" onClick={this.handleCondensationOff}>Heightened Rack View</Button>{' '}
-              <Button size="sm" onClick={this.showAllRacks}>Show All Racks</Button>{' '}
-              <Button size="sm" onClick={this.showLessRacks}>Show Less Racks</Button>{' '}
-            </ButtonGroup>
-          </ButtonToolbar>
-        </div>
-        <br></br>
-        <h1>Racks</h1>
-
-        {this.props.rack.map((item, key) =>
-          <div id="rackContainer">
-            <div id='hideOnPrint'>
-              {this.props.is_admin ? (
-                  <ButtonGroup>
-                    <Button color="warning" size="sm" onClick={() => this.showEditForm(item.id)}>Edit this
-                      Rack</Button>{' '}
-                    <Button color="danger" size="sm" onClick={() => this.showDeleteForm(item.id)}>Delete this
-                      Rack</Button>{' '}
-                  </ButtonGroup>
-                )
-                :
-                (<p></p>)}
-            </div>
-            <br></br>
-            <br></br>
-            <RackTable
-              sending={this.sendFromRow}
-              sendUrl={this.sendUrlInView}
-              rack={item}
-              condensedState={this.state.condensedView}
-              is_admin={this.props.is_admin}/>
-          </div>
+      <Container maxwidth="xl">
+          <Grid container className="themed-container" spacing={2}>
+            <Grid item justify="flex-start" alignContent='center' xs={12}/>
+            <Grid item justify="flex-start" alignContent='center' xs={2}>
+              {add}
+            </Grid>
+            <Grid item justify="center" alignContent="center" xs={3}>
+              {deleteMultiple}
+            </Grid>
+            <Grid item justify="center" alignContent="center" xs={3}>
+              {condensed}
+            </Grid>
+            <Grid item justify="center" alignContent="center" xs={3}>
+              {showAll}
+            </Grid>
+            <Grid item xs={12}>
+              {this.props.rack.map((item, key) =>
+              <div id="rackContainer">
+                <div id='hideOnPrint'>
+                  {this.props.is_admin ? (
+                      <ButtonGroup>
+                        <Button variant="outline" size="sm" onClick={() => this.showEditForm(item.id)} endIcon={<EditIcon />}></Button>{' '}
+                        <Button variant="outline" size="sm" onClick={() => this.showDeleteForm(item.id)} endIcon={<DeleteIcon />}></Button>{' '}
```

Comment:
For these, instead of using a Button with an EndIcon (which is the preferred implementation when a button has text and an icon), do you think you could use an IconButton? IconButtons are preferred when the button is just an icon, and should also include a ToolTip. Current implementation moves the icon to like the right side of the button (lopsided)
See here for reference:  https://material-ui.com/components/buttons/#icon-buttons. I also wonder if there's a way to make the delete button red...